### ISSUE TITLE
W5X00 and W6X00 series support in ROCm 5.X  GFX 1x00 Navi1 GFX 2x00 "big" navi

### ISSUE THREAD

**Issue by FCLC** [RadeonOpenCompute/ROCm#1617]
Without asking for a specific date, we've now seen 5.0 announced during the latest keynote, and we've also had our expectations set for  Navi 1 and Navi 2 support in the "early" 5.x series. What timeframe can we reasonably expect each of RDNA1 and RDNA2 to be enabled in the ROCm stack?

see #1595 , #1592 #1547 #1544 #1539 etc. 

I realize that writing a compute stack for a nascent language is no small feet, all the while trying to meet the deadlines for MI_2X0 support, and while getting the Frontier software stack ready. 

All the same, I don't really care about that, since I can't perform any initial development on my workstation before deploying to clusters such as frontier. 

We're over 2 years in for RDNA and this week marks 2 years since the W5700 first started shipping (and consequently landed on my desk). 

This isn't news to anyone, but a typical workflow is to develop on the workstation, train/compute in the cloud, then inference at the edge. 

We currently have no good way of doing  the first step outside of: 

A) Using aging hardware that becomes less and less relevant every quarter.
B) Putting an MI100 Server class card in a workstation chassis, 3d printing/manufacturing some sort of high airflow low noise bracket to force enough air through the card to keep it operating correctly. 

That or we use hip as an intermediary to cuda, at which point we may as well just be using DPCT/dpcpp from oneAPI and target anything from AVX to SYCL to FPGA etc.

**Comment by FCLC** [comment]
I'd also ask that this issue be left open until such a time as support is officially added to the ROCm stack.

**Comment by ROCmSupport** [comment]
Hi @FCLC 
Thanks for reaching out and sharing your views.
Navi21 support is on the way and its going to be officially supported very soon. Please stay tuned.
I am not closing this thread until Navi is enabled in ROCm officially.
Thank you.

**Comment by unexploredtest** [comment]
> I am not closing this thread until Navi is enabled in ROCm officially.

RDNA 1 included?

**Comment by FCLC** [comment]
> Navi21 support is on the way and its going to be officially supported very soon. Please stay tuned.
> 
> I am not closing this thread until Navi is enabled in ROCm officially.

@ROCmSupport I appreciate the response. 
Does the above include RDNA1? And if so, what is the general time frame? 

If I understand correctly the expected roadmap is: 
Finalize Navi21 -> Navi22 -> rest of Navi 2x series (Navi 23 and 24)-> Navi10 -> Navi 14/12

Please correct the above as needed.

Thanks, 

-Fclc

**Comment by ROCmSupport** [comment]
Hi @FCLC and @unexploredtest, thanks for your interest on Navi + ROCm.
I do not have concrete data on Navi roadmap. But I am sure that some set of Navi cards are coming with ROCm enabled very soon, I can not share all details right now. Request to wait for some time.
Thank you.

**Comment by FCLC** [comment]
> Hi @FCLC and @unexploredtest, thanks for your interest on Navi + ROCm. I do not have concrete data on Navi roadmap. But I am sure that some set of Navi cards are coming with ROCm enabled very soon, I can not share all details right now. Request to wait for some time. Thank you.

Thanks for this. I unfortunately haven't seen any mention of Navi 1. 

Would it be possible to mark/use this issue to log which graphics core's have and have not received support?

**Comment by FCLC** [comment]
Looks like the next release that will expand support for RDNA is coming in early 2022 per #1618

**Comment by Djip007** [comment]
https://rocmdocs.amd.com/en/latest/Installation_Guide/Installation_new.html#confirm-you-have-a-rocm-capable-gpu
the RNDA2 AMD Radeon™ PRO W6800 is list as supported
is with release 4.5.2 RDNA2 in some card enable ???

**Comment by FCLC** [comment]
> the RNDA2 AMD Radeon™ PRO W6800 is list as supported
> is with release 4.5.2 RDNA2 in some card enable ???

Yes, as of 4.5.x, the W6800 (and to my knowledge it's consumer counter part) have near if not complete support within ROCm. 

The underlying GPU die, Navi21, has different sub versions beyond the Main die in the W6800. 

The sub versions can be found here: [off site link](https://www.techpowerup.com/gpu-specs/amd-navi-21.g923)

**Comment by Djip007** [comment]
https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/issues/173#issuecomment-991679054
I did some test on a 6900XT... can confirm (if neaded) rocm / tensorflow is enable (test with rocm/tensorflow-autobuilds container...)

**Comment by FCLC** [comment]
> I did some test on a 6900XT... can confirm (if neaded) rocm / tensorflow is enable

That would be superb if possible!

**Comment by FCLC** [comment]
@ROCmSupport Any movement on this? Should we expect movement this quarter?

**Comment by ROCmSupport** [comment]
Hi @FCLC 
I am sure that some set of Navi series cards are already enabled and it will be officially supported in/from this quarter. Please stay tuned for our latest updates via our documentation and release notes. Thank you.

**Comment by FCLC** [comment]
From the updated documentation in the [ROCm 5.0 guide](https://docs.amd.com/bundle/ROCm_Installation_Guide-v5.0/page/Prerequisite_Actions.html#d3848e335) it seems that there has't been much movement on expanding the support lineup. 

Seems that 5.0 added official support for Radeon Pro V620 and W6800 Workstation GPUs.

**Comment by gobenji** [comment]
> I did some test on a 6900XT... can confirm (if neaded) rocm / tensorflow is enable (test with rocm/tensorflow-autobuilds container...)

I tried with a 6700XT. OpenCL works but not tensorflow.

In rocm/tensorflow-autobuilds:
```
root@10befa294fc0:/workspace# python3
Python 3.9.10 (main, Jan 15 2022, 18:56:52) 
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
>>> tf.__version__
'2.9.0'
>>> tf.reduce_sum(tf.random.normal([1000, 1000]))
"hipErrorNoBinaryForGpu: Unable to find code object for all current devices!"
```

However it is still based on rocm-4.5.2

**Comment by FCLC** [comment]
> However it is still based on rocm-4.5.2

if you could try again with the new release that would be of some help!

**Comment by gobenji** [comment]
> if you could try again with the new release that would be of some help!

Same result with the newer image.
Looking into the image, I'm not sure but I guess this is the list of available "code objects" that tensorflow is looking for:
```
/opt/rocm-5.0.0/lib/library/Kernels.so-000-gfx1010.hsaco         
/opt/rocm-5.0.0/lib/library/Kernels.so-000-gfx1011.hsaco              
/opt/rocm-5.0.0/lib/library/Kernels.so-000-gfx1012.hsaco              
/opt/rocm-5.0.0/lib/library/Kernels.so-000-gfx1030.hsaco                
/opt/rocm-5.0.0/lib/library/Kernels.so-000-gfx803.hsaco                 
/opt/rocm-5.0.0/lib/library/Kernels.so-000-gfx900.hsaco                      
/opt/rocm-5.0.0/lib/library/Kernels.so-000-gfx906-xnack-.hsaco         
/opt/rocm-5.0.0/lib/library/Kernels.so-000-gfx908-xnack-.hsaco         
/opt/rocm-5.0.0/lib/library/Kernels.so-000-gfx90a-xnack+.hsaco         
/opt/rocm-5.0.0/lib/library/Kernels.so-000-gfx90a-xnack-.hsaco
```
[chip names - product names](https://llvm.org/docs/AMDGPUUsage.html#processors)
So maybe it will work on 6800, 6800xt and 6900xt gpus (gfx1030).

**Comment by xuhuisheng** [comment]
@gobenji RX6700XT may be gfx1031, you could check it by `/opt/rocm/bin/rocminfo`.

**Comment by jasondrusso** [comment]
I tested ROCm 5.0 with a 6800xt, and I'm still seeing the 'hipErrorNoBinaryForGpu: Unable to find code object for all current devices' error.

**Comment by FCLC** [comment]
> I tested ROCm 5.0 with a 6800xt, and I'm still seeing the 'hipErrorNoBinaryForGpu: Unable to find code object for all current devices' error.

Had you fully purged ROCm 4.x? that could be a factor

**Comment by jasondrusso** [comment]
Yes, I'm pretty sure I removed everything from 4.x before I installed 5.0

**Comment by FCLC** [comment]
> Yes, I'm pretty sure I removed everything from 4.x before I installed 5.0. I removed the old amd drivers to upgrade to 21.50 & also removed the 4.x repos.

Interesting, since it's fundamentally the same gpu it should be able to deal with the same compute kernels. I'd recommend opening a new issue and then referencing this issue.

**Comment by jasondrusso** [comment]
Ok thanks. But before I do, does it matter that I'm trying this with Tensorflow 1.15 and python 3.7?

**Comment by FCLC** [comment]
> Ok thanks. But before I do, does it matter that I'm trying this with Tensorflow 1.15 and python 3.7? 

You would have to double check the docs. Beyond that, I'm more on the HPC side than ai side. Others may be better equipped to answer that question.

**Comment by gobenji** [comment]
> Ok thanks. But before I do, does it matter that I'm trying this with Tensorflow 1.15 and python 3.7?

How did you try those versions with rocm 5.0.0? The versions in the rocm/tensorflow-autobuilds image are python 3.9.10 and tensorflow 2.9.0.

What result do you get if you use the docker image directly?
```
docker run --rm --device=/dev/kfd --device=/dev/dri --security-opt seccomp=unconfined rocm/tensorflow-autobuilds python3 -c "import tensorflow as tf; print(tf.reduce_sum(tf.random.normal([1000, 1000])))"
```

**Comment by jasondrusso** [comment]
I outlined how I set up & tested rocm 5.0 with Tensorflow 1 here: https://github.com/RadeonOpenCompute/ROCm/issues/1676

I don't have docker, but when I run the test above with a Python 3.9 venv & Tensorflow 2.8.0 (latest version I can find in pip), I get the following error:

2022-02-14 11:14:10.639407: I tensorflow/stream_executor/rocm/rocm_gpu_executor.cc:832] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2022-02-14 11:14:10.639462: E tensorflow/stream_executor/rocm/rocm_driver.cc:1196] failed to query max grid dim x: HIP_ERROR_InvalidValue
2022-02-14 11:14:10.639481: F tensorflow/core/platform/statusor.cc:33] Attempting to fetch value instead of handling error NOT_FOUND: could not retrieve ROCM device attribute (7): HIP_ERROR_InvalidValue

**Comment by FCLC** [comment]
Tho this is far from official support, this repository may be of some use:

https://github.com/xuhuisheng/rocm-build

It purports to have found ways to add compilation targets for non-officially supported GPU's. I'd personally caution anyone considering using this to limit its use to development environments only and not production machines. 

I'm currently busy preparing for AVX512FP16 and AMX BF16/INT8 kernels so don't have time to test in the immediate.

If anyone can confirm if the above does or does not work, that would be superb. 

In the meantime hoping to see an update from the ROCm team soon :sweat_smile:

**Comment by ROCmSupport** [comment]
Hi All,
Thanks you.
As Radeon Pro W6800 enabled(Navi21 workstation card) from ROCm 5.0 onwards, I am closing this thread.
Hope this helps.
Please let me know if you need more information and request to file new tickets for quick resolutions.
Thank you.

**Comment by jasondrusso** [comment]
> > 
> 
> Hi, Since this thread was about all of Navi, including generation 1, I do not think the above fits the criteria for closing this issue.
> 
> As of now only 1 GPU sub-group amongst both RDNA generations has received support, that being Navi 21 (and by extension it's consumer equivalents)
> 
> This leaves those running other RDNA workstation class cards such as the: W6600 (Navi 23) W6400 (Navi 24) W5700 (Navi 10) W5500 (Navi 14) Radeon Pro v520 (Navi 12)
> 
> and all of their mobile equivalents out to dry.
> 
> Outside of the above cores, Navi 22 does not have a workstation equivalent, but is worth taking note of.

As 'Navi' is a pretty large category of products, with new ones being added, I wonder if it would be better to close this issue and open new issues for each specific Navi group. At the very least, it might make tracking support amongst the product lines easier.

**Comment by FCLC** [comment]
@ROCmSupport in reference to https://github.com/RadeonOpenCompute/ROCm/issues/1617#issuecomment-1047000685

And 
https://github.com/RadeonOpenCompute/ROCm/issues/1617#issuecomment-1049297130

Would it be better to open a new issue for each of the sub groups of cards still missing support or reopen this issue? 

I'm happy to see that some progress was made by introducing support for Navi 21, but want to make sure that other cards aren't left to fall to the wayside.

**Comment by felixniemeyer** [comment]
I sold my 5700 some time ago because ROCm support would just never come...
It's sad, but I realize, there are just too few people who want to do 
- compute stuff (machine learning, video editing, maybe mining for fun) 
- on a AMD GPU instead of getting a NVIDIA card (modern and < 900USD) 
- on Linux
It seems to me, that I have to give up one of these three things else I'd wait forever... :-(

**Comment by FCLC** [comment]
> I sold my 5700 some time ago because ROCm support would just never come...
> 
> It's sad, but I realize, there are just too few people who want to do 
> 
> - compute stuff (machine learning, video editing, maybe mining for fun) 
> 
> - on a AMD GPU instead of getting a NVIDIA card (modern and < 900USD) 
> 
> - on Linux
> 
> 
> 
> It seems to me, that I have to give up one of these three things else I'd wait forever... :-(

Actually, after having spent ~a dozen hours building the entire stack, it's seems like gfx1010/Navi10 cards are usable, even if the work needed to do so it somewhat ridiculous

**Comment by maxap** [comment]
> 

what did you do to make it work? I'm in need for this

**Comment by jasondrusso** [comment]
Rocm 5.4.1 should work pretty good with navi now. I have a 6800xt and using the amdgpu-install script installed easily when run with rocm usecase. However, to use with tensorflow, there are some extra packages you might have to install (I think I have to add rocm-smi, hipfft, miopen and one or two others) to get that to work with rocm (not to mention pip install tensorflow-rocm).
